### PR TITLE
Handle optional arg

### PR DESCRIPTION
**Pull request by wsaihopfsg**
In `request_handle_udp`, it should be possible to provide the optional `start_angle` & `max_num_points_scan` without having the need to go through dynamic reconfiguration.

This PR would allow the user to `roslaunch` with these 2 arguments, for example
```
roslaunch pf_driver R2300.launch start_angle:=-10 max_num_points_scan:=101
```
If any of these parameters are not provided, the default values will be used. Hence invoking the old way
```
roslaunch pf_driver R2300.launch
```
would start the package with `start_angle=-180` & `max_num_points_scan=0` i.e. no change in behaviour.

Note that the code changes has only been tested with R2300 and not with R2000.

### CODE REVIEW

**Comment by ipa-hsd** [comment]
@wsaihopfsg thanks for the contribution! Could you please resolve the conflicts?

**Comment by wsaihopfsg** [comment]
> @wsaihopfsg thanks for the contribution! Could you please resolve the conflicts?

I have resolved the conflicts in this PR. May be you should check and merge this first before I resolve https://github.com/PepperlFuchs/ROS_driver/pull/49 & https://github.com/PepperlFuchs/ROS_driver/pull/50?

**Comment by ipa-hsd** [comment]
@wsaihopfsg sorry I thought I submitted these comments already. Just saw that they were still pending. I will come up with a solution today and then you can take a look.

File: pf_driver/include/pf_driver/pf/pf_interface.h [hunk 1]
```diff
@@ -37,48 +39,50 @@ class PFInterface
...
+    PFState state_;
+    std::string product_;
+    std::string expected_device_;
 
-  HandleInfo info_;
-  ScanConfig config_;
-  ScanParameters params_;
+    HandleInfo info_;
+    ScanConfig config_;
+    ScanParameters params_;
```
Comment (ipa-hsd) [inline pf_driver/include/pf_driver/pf/pf_interface.h:69]: Please take care with the formatting. Whitespaces should not be introduced unnecessarily.

**Comment by ipa-hsd** [comment]
Closing this for https://github.com/PepperlFuchs/ROS_driver/pull/51

**Pull request closed without merging**